File: src/main/java/de/tectoast/emolga/commands/music/control/DcCommand.java
package de.tectoast.emolga.commands.music.control;

import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.commands.MusicCommand;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.TextChannel;

public class DcCommand extends MusicCommand {
    public DcCommand() {
        super("dc", "Lässt den Bot disconnecten");
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        TextChannel tco = e.getChannel();
        Message m = e.getMessage();
        String msg = m.getContentDisplay();
        if (tco.getGuild().getId().equals("447357526997073930")) {
            e.getJDA().getGuildById(msg.substring(4)).getAudioManager().closeAudioConnection();
            return;
        }
        tco.getGuild().getAudioManager().closeAudioConnection();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/music/control/DcCommand.kt
package de.tectoast.emolga.commands.music.control

import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.commands.MusicCommand

class DcCommand : MusicCommand("dc", "Lässt den Bot disconnecten") {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override fun process(e: GuildCommandEvent) {
        val tco = e.textChannel
        val m = e.message!!
        val msg = m.contentDisplay
        if (tco.guild.id == "447357526997073930") {
            e.jda.getGuildById(msg.substring(4))!!.audioManager.closeAudioConnection()
            return
        }
        tco.guild.audioManager.closeAudioConnection()
    }
}